learn-data-structures-by-building-a-shopping-cart/index19.js

```diff
@@ -0,0 +1,67 @@
+    
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+    this.taxRate = 8.25;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+  
+  applyDiscount(amount) {
+    return parseFloat((this.discountPercentage / 100 * amount).toFixed(2));
+  }
+}
+
+/*
+Now create a function called calculateTaxes.
```

Comment:
No need to break this one up since it's very similar to the last thing the student did. Here's some updated text:

Now create a function called `calculateTaxes` that takes `amount` as a parameter and returns the tax due.
The formula is the same as the previous `applyDiscount` function.
Take the tax rate, divide it by 100 then multiply by the provided amount.
Then round up to 2 decimal points and convert it back to number before returning.